From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Li Hua Qian <huaqian.li@siemens.com>
Date: Mon, 4 Nov 2024 10:46:14 +0800
Subject: [PATCH] driver: iot2050: Add a temporary workaround for the eMMC
 issue

After approximately 2 months of operation, the device may fail to detect
the eMMC when rebooting. This commit provides a temporary workaround to
bypass the eMMC issue.

TODO: Implement a permanent solution for eMMC detection issues.

Signed-off-by: Li Hua Qian <huaqian.li@siemens.com>
---
 drivers/mmc/mmc.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/mmc.c b/drivers/mmc/mmc.c
index 31cfda288587..dff0ff89a801 100644
--- a/drivers/mmc/mmc.c
+++ b/drivers/mmc/mmc.c
@@ -729,8 +729,16 @@ static int mmc_send_op_cond(struct mmc *mmc)
 		if (mmc->ocr & OCR_BUSY)
 			break;
 
-		if (get_timer(start) > timeout)
-			return -ETIMEDOUT;
+		/*
+		 * TODO: Implement a permanent solution for the eMMC detection issues.
+		 * After approximately 2 months of operation, the device may fail to
+		 * detect the eMMC when rebooting. Here provides a temporary workaround
+		 * to bypass the issue.
+		 */
+		if (get_timer(start) > timeout) {
+			printf("mmc: eMMC is busy, continue booting\n");
+			break; //return -ETIMEDOUT;
+		}
 		udelay(100);
 	}
 	mmc->op_cond_pending = 1;
